File: 3. Arrays/1. Easy/Java/12. Find the element that appears once and others twice in an array.java
/* package codechef; // don't place package name! */

import java.util.*;
import java.lang.*;
import java.io.*;

/* Name of the class has to be "Main" only if the class is public. */

/*
Find the element that appears once and others twice in an array
*/

class Codechef
{
    
    public static int findSingleElement(int arr[],int n){
        int xor1 = 0;
        for(int i=0;i<n;i++){
            xor1 = xor1 ^ arr[i];
        }
        return xor1;
    }
    
	public static void main (String[] args) throws java.lang.Exception
	{
		// your code goes here
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();
		int arr[] = new int[n];
		for(int i=0;i<n;i++){
		    arr[i] = sc.nextInt();
 		}
 		int singleElement = findSingleElement(arr,n);
 		System.out.println(singleElement);
	}
}

File: 3. Arrays/1. Easy/C++/12. Find the element that appears once and others twice in an array.cpp
/*
Find the element that appears once and others twice in an array
*/

#include <bits/stdc++.h>
using namespace std;


int findSingleElement(int arr[],int n){
    int xor1 = 0;
    for(int i=0;i<n;i++){
        xor1 = xor1 ^ arr[i];
    }
    return xor1;
}

int main() {
    //Write your code here
    int n;
    cin >> n;
    int arr[n];
    for(int i=0;i<n;i++){
        cin >> arr[i];
    }
    int singleElement = findSingleElement(arr,n);
    cout << singleElement;
    return 0;
}